packer/windows.pkr.hcl

```diff
@@ -0,0 +1,189 @@
+
+      "choco install -y --no-progress 7zip",
+      "choco install -y --no-progress git --parameters=\"/GitAndUnixToolsOnPath\"",
+    ]
+    only = ["googlecompute.windows-ci-mingw64", "googlecompute.windows-ci-vs-2019"]
+  }
+
+  provisioner "powershell" {
+    inline = [
+      # contains useful utilities, including a diff we can use
+      "[Environment]::SetEnvironmentVariable('PATH',  'C:\\Program Files\\Git\\usr\\bin;' + [Environment]::GetEnvironmentVariable('PATH', 'Machine'), 'Machine')",
+    ]
+  }
+
+  # install windows debugger
+  provisioner "powershell" {
+    script = "scripts/windows_install_dbg.ps1"
+  }
+
+  # install python
+  provisioner "powershell" {
+    environment_vars = ["TEMP_PYTHON_VERSION=3.10.6"]
+    script = "scripts/windows_install_python.ps1"
+  }
+
+  # install meson and ninja
+  provisioner "powershell" {
+    inline = ["py -m pip install meson ninja"]
+  }
+
+  # install perl
+  provisioner "powershell" {
+    environment_vars = ["TEMP_PERL_VERSION=5.26.3.1"]
+    script = "scripts/windows_install_perl.ps1"
+  }
+
+  # install openssl
+  provisioner "powershell" {
+    script = "scripts/windows_install_openssl.ps1"
+  }
+  ### end of base installations
+
+  ### mingw installations
+  provisioner "powershell" {
+    inline = [
+      "[Environment]::SetEnvironmentVariable('MSYSTEM', 'UCRT64', 'Machine')",
+      # this could be reduntant
+      "$env:MSYSTEM = 'UCRT64'",
+    ]
+    only = ["googlecompute.windows-ci-mingw64", "docker.windows-ci-mingw64-docker"]
+  }
+
+  provisioner "powershell" {
+    script = "scripts/windows_install_mingw64.ps1"
+    only = ["googlecompute.windows-ci-mingw64", "docker.windows-ci-mingw64-docker"]
+  }
+
+  # Change default console code page (0) with Windows code page (65001) to get rid of warnings in postgres tests
+  provisioner "powershell" {
+    inline = ["chcp 65001"]
+    only = ["googlecompute.windows-ci-mingw64", "docker.windows-ci-mingw64-docker"]
+  }
+  ### end of mingw installations
+
+  ### vs-2019 installations
+  provisioner "powershell" {
+    script = "scripts/windows_install_winflexbison.ps1"
+    only = ["googlecompute.windows-ci-vs-2019", "docker.windows-ci-vs-2019-docker"]
+  }
+
+  provisioner "powershell" {
+    script = "scripts/windows_install_pg_deps.ps1"
+    only = ["googlecompute.windows-ci-vs-2019", "docker.windows-ci-vs-2019-docker"]
+  }
+
+  provisioner "powershell" {
+    script = "scripts/windows_install_vs_2019.ps1"
+    only = ["googlecompute.windows-ci-vs-2019", "docker.windows-ci-vs-2019-docker"]
+  }
+  ### end of vs-2019 installations
+
+  post-processors {
+    post-processor "docker-tag" {
+        repository =  "${var.docker_repo}/${var.task_name}"
+        tags = ["latest"]
+        only = ["docker.windows-ci-vs-2019-docker", "docker.windows-ci-mingw64-docker"]
+      }
+    post-processor "docker-push" {
+      login = true
+      login_username = "oauth2accesstoken"
```

Comment:
Might be good to reference wherever you found this...

Reply:
Done.